app/controllers/orders_controller.rb

```diff
@@ -2,10 +2,21 @@ class OrdersController < ApplicationController
   before_action :find_order, only: [:cart, :checkout, :update, :receipt]
   before_action :check_access, only: [:cart, :checkout, :receipt]
 
+  def add_to_cart
+    product = Product.find(params[:id])
```

Comment:
this should have access to @order; add #add_to_cart to the before_action :find_order